pallets/runtime/src/bridge.rs

```diff
@@ -0,0 +1,376 @@
+    /// of required signatures.
+    pub nonce: u64,
+    /// The signers of the multisig.
+    pub signers: Vec<Signatory>,
+    /// The number of required signatures in the multisig.
+    pub signatures_required: u64,
+}
+
+/// Information about a change of the validator set including the multisig account address of the
+/// new validator set and the accepted proposal.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct ValidatorSetChange<AccountId> {
+    /// The new multisig account address.
+    pub account_id: AccountId,
+    /// The accepted proposal.
+    pub validator_set: ValidatorSet,
+}
+
+/// The intended recipient of POLY exchanged from the locked ERC20 tokens.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum IssueRecipient<AccountId> {
+    Account(AccountId),
+    Identity(IdentityId),
+}
+
+/// A unique lock-and-mint bridge transaction containing Ethereum transaction data and a bridge nonce.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct BridgeTx<AccountId> {
+    /// Bridge validator runtime nonce.
+    pub nonce: u64,
+    /// Recipient of POLY on Polymesh: the deposit address or identity.
+    pub recipient: IssueRecipient<AccountId>,
+    /// Amount of tokens locked on Ethereum.
+    pub value: u128,
+    /// Ethereum token lock transaction hash.
+    pub tx_hash: H256,
+}
+
+/// A transaction that is pending a valid identity KYC.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct PendingTx<AccountId> {
+    /// The identity on which the KYC is pending.
+    pub did: IdentityId,
+    /// The pending transaction.
+    pub bridge_tx: BridgeTx<AccountId>,
+}
+
+/// Either a pending transaction or a `None` or an error.
+type IssueResult<AccountId> = sp_std::result::Result<Option<PendingTx<AccountId>>, DispatchError>;
+
+decl_error! {
+    pub enum Error for Module<T: Trait> {
+        /// The bridge validator set address is not set.
+        ValidatorsNotSet,
+        /// The proposal doesn't contain any signers.
+        NoSignersProvided,
+        /// The proposed number of required signatures is out of bounds.
+        RequiredSignaturesOutOfBounds,
+        /// The validator does not have an identity.
+        IdentityMissing,
+        /// Failure to credit the recipient account.
+        CannotCreditAccount,
+        /// Failure to credit the recipient identity.
+        CannotCreditIdentity,
+        /// The origin is not the validator set multisig.
+        BadCaller,
+        /// The recipient DID has no valid KYC.
+        NoValidKyc,
+    }
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as Bridge {
+        /// The multisig account of the bridge validator set. The genesis signers must accept their
+        /// authorizations to be able to get their proposals delivered.
+        Validators get(validators) build(|config: &GenesisConfig| {
```

Comment:
Let's rename `Validators` to something else to avoid confusion with the block validators.

Reply:
How about `BridgeSigners`?